packages/plugins/whisper/src/blockchain.js

```diff
@@ -0,0 +1,435 @@
+    this.logger.error("===============================================================================".yellow);
+    this.logger.error("===========> ".yellow + __('WARNING! ÐApp path length is too long: ').yellow + _dappPath.yellow);
+    this.logger.error("===========> ".yellow + __('This is known to cause issues with starting geth, please consider reducing your ÐApp path\'s length to 66 characters or less.').yellow);
+    this.logger.error("===============================================================================".yellow);
+  }
+};
+
+Blockchain.prototype.isClientInstalled = function (callback) {
+  let versionCmd = this.client.determineVersionCommand();
+  this.runCommand(versionCmd, {}, (err, stdout, stderr) => {
+    if (err || !stdout || stderr.indexOf("not found") >= 0 || stdout.indexOf("not found") >= 0) {
+      return callback(__('Ethereum client bin not found:') + ' ' + this.client.getBinaryPath());
+    }
+    const parsedVersion = this.client.parseVersion(stdout);
+    const supported = this.client.isSupportedVersion(parsedVersion);
+    if (supported === undefined) {
+      this.logger.error((__('WARNING! Ethereum client version could not be determined or compared with version range') + ' ' + this.client.versSupported + __(', for best results please use a supported version')).yellow);
+    } else if (!supported) {
+      this.logger.error((__('WARNING! Ethereum client version unsupported, for best results please use a version in range') + ' ' + this.client.versSupported).yellow);
+    }
+    callback();
+  });
+};
+
+Blockchain.prototype.initDevChain = function(callback) {
+  const self = this;
+  const ACCOUNTS_ALREADY_PRESENT = 'accounts_already_present';
+  // Init the dev chain
+  self.client.initDevChain(self.config.datadir, (err) => {
+    if (err) {
+      return callback(err);
+    }
+
+    const accountsToCreate = self.config.account && self.config.account.numAccounts;
+    if (!accountsToCreate) return callback();
+
+    // Create other accounts
+    async.waterfall([
+      function listAccounts(next) {
+        self.runCommand(self.client.listAccountsCommand(), {}, (err, stdout, _stderr) => {
+          if (err || stdout === undefined || stdout.indexOf("Fatal") >= 0) {
+            console.log(__("no accounts found").green);
+            return next();
+          }
+          // List current addresses
+          self.config.unlockAddressList = self.client.parseListAccountsCommandResultToAddressList(stdout);
+          // Count current addresses and remove the default account from the count (because password can be different)
+          let addressCount = self.config.unlockAddressList.length;
+          if (addressCount < accountsToCreate) {
+            next(null, accountsToCreate - addressCount);
+          } else {
+            next(ACCOUNTS_ALREADY_PRESENT);
+          }
+        });
+      },
+      function newAccounts(accountsToCreate, next) {
```

Comment:
Are operations like `newAccounts` and `listAccounts` needed for the whisper client? On `master` currently, they're effectively turned into no-ops and I introduced [some changes](https://github.com/embark-framework/embark/pull/2034) related to that. I'm wondering if this waterfall could be simplified further.